vita3k/modules/SceCtrl/SceCtrl_tracy.cpp

```diff
@@ -0,0 +1,578 @@
+// Vita3K emulator project
+// Copyright (C) 2022 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+/**
+ * @file SceCtrl_tracy.cpp
+ * @brief Tracy argument serializing functions for SceCtrl
+ */
+
+#include "ctrl/ctrl.h"
+#include <sstream>
+#include <string>
+#ifdef TRACY_ENABLE
+
+#include "SceCtrl_tracy.h"
+
+void tracy_sceCtrlGetControllerPortInfo(tracy::ScopedZone *___tracy_scoped_zone, bool module_activation, SceCtrlPortInfo *info) {
+    // Tracy activation state check to log arguments
+    if (!module_activation)
+        return;
+
+    struct TracyLogSettings {
+        struct args {
+            std::string info = "&info: ";
+        } args;
+
+    } tracy_settings;
+
+    // Log enumerated args
+
+    std::stringstream infoss;
+    infoss << info;
+    tracy_settings.args.info += infoss.str();
+
+    // Send args
+    ___tracy_scoped_zone->Text(tracy_settings.args.info.c_str(), tracy_settings.args.info.size());
```

Comment:
I'm not sure, but can't all of this be simplified into:
```
const std::string infos = fmt::format("&info : {}", log_hex(static_cast<uin64_t>(info)));
___tracy_scoped_zone->Text(infos.c_str(), infos.size()); 
```
Moreoever, right now on Tracy it shows something like `&info: 000000040154FA6C`. Printing the host pointer is not really useful, the least that can be done would be to print the guest pointer (or better the content of the struct even though it would be a bit harder to code).

Reply:
the goal of printing the host pointer was to check in case its an invalid pointer, like 0 or 4, which are somewhat common, i would mainly print pointers in case they are out pointers. also yeah you are right, it can be simpler, i was just following the way it was first implemented